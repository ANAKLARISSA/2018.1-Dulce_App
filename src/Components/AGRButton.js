import React from 'react';
import {TouchableHighlight,Text,View} from 'react-native';

const styles = {
  container: {
    backgroundColor: 'grey',
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 30,
    paddingHorizontal: 50
  },
  text: {
    color: 'white',
    fontSize: 18
  },
  view: {
    paddingHorizontal: 30
  }
};

class AGRButton extends React.Component {
  render() {
    return (
      <View style={styles.view}>
      <TouchableHighlight
        style={styles.container}
        onPress={() => this.props.onPress()}
      >
        <Text style={styles.text}>{this.props.text}</Text>
      </TouchableHighlight>
      </View>
    );
  }
}

export default AGRButton;
