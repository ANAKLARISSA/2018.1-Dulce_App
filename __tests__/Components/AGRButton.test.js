import React from 'react';
import AGRButton from '../../src/Components/AGRButton';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
require('bezier');

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
  .create(<AGRButton />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Testing onPress', () => {
  const onPressMock = jest.fn();
  const wrapper = shallow(<AGRButton onPress = {onPressMock}/>);
  const button = wrapper.find('Styled(Button)').at(0);
  button.simulate('press');
  expect(onPressMock).toBeCalled();
});
