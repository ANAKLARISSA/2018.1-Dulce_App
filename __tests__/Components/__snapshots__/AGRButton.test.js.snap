// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  viewIsInsideTabBar={false}
  virtual={undefined}
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#5f4b8b",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 57,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 2,
          "flex": 8,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 0,
          "marginTop": 30,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "width": 180,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "center",
            "color": "white",
            "fontFamily": "Raleway",
            "fontSize": 16,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
